src/components/Preferences/Preferences.js

```diff
@@ -0,0 +1,218 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Transition, animated } from 'react-spring'
+import {
+  AppBar,
+  AppView,
+  BREAKPOINTS,
+  ButtonIcon,
+  IconClose,
+  TabBar,
+  Viewport,
+  breakpoint,
+  font,
+  springs,
+} from '@aragon/ui'
+import LocalIdentitiesComponent from './LocalIdentities'
+import { AragonType } from '../../prop-types'
+import {
+  IdentityContext,
+  identityEventTypes,
+} from '../IdentityManager/IdentityManager'
+
+const TABS = ['Network', 'Manage labels']
+const ESCAPE_KEY_CODE = 27
+
+const Preferences = ({ onClose, smallView, wrapper }) => {
+  const { identityEvents$ } = React.useContext(IdentityContext)
+  const [selectedTab, setSelectedTab] = React.useState(1)
+  const [localIdentities, setLocalIdentities] = React.useState({})
+  const handleGetAll = async () => {
+    if (!wrapper) {
+      return
+    }
+    setLocalIdentities(await wrapper.getLocalIdentities())
+  }
+  const handleClearAll = async () => {
+    if (!wrapper) {
+      return
+    }
+    await wrapper.clearLocalIdentities()
+    setLocalIdentities({})
+    identityEvents$.next({ type: identityEventTypes.CLEAR })
+  }
+  const handleModify = (address, data) => {
+    if (!wrapper) {
+      return
+    }
+    wrapper.modifyAddressIdentity(address, data)
+  }
+  const handleImport = async list => {
+    if (!wrapper) {
+      return
+    }
+    setLocalIdentities({})
+    for (const { name, address } of list) {
+      await wrapper.modifyAddressIdentity(address, { name })
+    }
+    setLocalIdentities(await wrapper.getLocalIdentities())
+    identityEvents$.next({ type: identityEventTypes.IMPORT })
+  }
+  const handleKeyUp = e => {
+    if (e.keyCode === ESCAPE_KEY_CODE) {
+      onClose()
+    }
+  }
+  React.useEffect(() => {
+    handleGetAll()
+    window.addEventListener('keyup', handleKeyUp)
```

Comment:
Can we use the `keydown` event for this? It’s usually what OSes do (and what we do in other places too).